vortex-ipc/src/messages.rs

```diff
@@ -186,15 +192,120 @@ impl<'a> WriteFlatBuffer for IPCArray<'a> {
             .collect_vec();
         let children = Some(fbb.create_vector(&children));
 
-        fba::Array::create(
+        let stats = compute_and_build_stats(fbb, self.1);
+
+        Array::create(
             fbb,
-            &fba::ArrayArgs {
+            &ArrayArgs {
                 version: Default::default(),
                 has_buffer: column_data.buffer().is_some(),
                 encoding,
                 metadata,
+                stats: Some(stats),
                 children,
             },
         )
     }
 }
+
+/// Computes all stats and uses the results to create an ArrayStats table for the flatbuffer message
+fn compute_and_build_stats<'a>(
```

Comment:
This doesn't let the writer configure which stats are written. Can you think of a clean way to support that?

Reply:
I don't think that's right. ViewContext represents the context required to interpret an array view at read-time. The stats configuration is not part of that.

I would actually say this isn't the job of this file. This file can just serialise the array and any stats that it has _already computed_. The caller (in this case, the IPCWriter) can set/clear whatever stats it likes before serialisation.